src/web/components/ApiKeyManagement/KeyEditDialog.tsx

```diff
@@ -23,6 +23,15 @@ type KeyEditDialogProps = {
   setApiKey: React.Dispatch<React.SetStateAction<ApiKeyDTO>>;
 };
 
+function getAllowedRoles(apiRoleAllowedLists: ApiRoleDTO[][]): ApiRoleDTO[] {
```

Comment:
Is it worth adding some basic unit tests for this?

Reply:
Sounds good. I'll put one in.